src/main/kotlin/kotlinx/kover/api/KoverConfig.kt

```diff
@@ -0,0 +1,541 @@
+     */
+    public fun htmlReport(config: Action<KoverProjectHtmlConfig>) {
+        config.execute(htmlReport)
+    }
+
+    /**
+     * Configures the verification task, including adding verification rules and whether it should be
+     * verified during the 'check' task.
+     */
+    public fun verify(config: Action<KoverVerifyConfig>) {
+        config.execute(verify)
+    }
+
+
+    // DEPRECATIONS
+    // TODO delete in 0.7 version
+    @Suppress("DEPRECATION")
+    @get:Internal
+    @Deprecated(
+        message = "Property was removed in Kover API version 2. Please refer to migration guide in order to migrate: ${KoverMigrations.MIGRATION_0_5_TO_0_6}",
+        replaceWith = ReplaceWith("engine"),
+        level = DeprecationLevel.ERROR
+    )
+    public val coverageEngine: Property<CoverageEngine> = objects.property(CoverageEngine::class.java)
+
+    @get:Internal
+    @Deprecated(
+        message = "Property was removed in Kover API version 2. Please refer to migration guide in order to migrate: ${KoverMigrations.MIGRATION_0_5_TO_0_6}",
+        replaceWith = ReplaceWith("engine"),
+        level = DeprecationLevel.ERROR
+    )
+    public val intellijEngineVersion: Property<String> = objects.property(String::class.java)
+
+    @get:Internal
+    @Deprecated(
+        message = "Property was removed in Kover API version 2. Please refer to migration guide in order to migrate: ${KoverMigrations.MIGRATION_0_5_TO_0_6}",
+        replaceWith = ReplaceWith("engine"),
+        level = DeprecationLevel.ERROR
+    )
+    public val jacocoEngineVersion: Property<String> = objects.property(String::class.java)
+
+    @get:Internal
+    @Deprecated(
+        message = "Property was removed in Kover API version 2. Please refer to migration guide in order to migrate: ${KoverMigrations.MIGRATION_0_5_TO_0_6}",
+        level = DeprecationLevel.ERROR
+    )
+    public var generateReportOnCheck: Boolean = true
+
+    @get:Internal
+    @Deprecated(
+        message = "Property was removed in Kover API version 2. Please refer to migration guide in order to migrate: ${KoverMigrations.MIGRATION_0_5_TO_0_6}",
+        level = DeprecationLevel.ERROR
+    )
+    public var disabledProjects: Set<String> = emptySet()
+
+    @Deprecated(
+        message = "Property was removed in Kover API version 2. Please refer to migration guide in order to migrate: ${KoverMigrations.MIGRATION_0_5_TO_0_6}",
+        level = DeprecationLevel.ERROR
+    )
+    @get:Internal
+    public var instrumentAndroidPackage: Boolean = false
+
+    @Deprecated(
+        message = "Property was removed in Kover API version 2. Please refer to migration guide in order to migrate: ${KoverMigrations.MIGRATION_0_5_TO_0_6}",
+        level = DeprecationLevel.ERROR
+    )
+    @get:Internal
+    public var runAllTestsForProjectTask: Boolean = false
+}
+
+// DEPRECATIONS
+// TODO delete in 0.7 version
+@Deprecated(
+    message = "Class was renamed in Kover API version 2",
```

Comment:
I've tried to migrate coroutines. Alas, tooling is far from being perfect -- the project cannot be imported into IDEA, and thus replacements have no use.

I suggest being more detailed in messages (that one can see from IDEA import).
For example, in this particular case, I suggest using `KoverExtension was renamed to KoverProjectConfig in Kover API version 2` otherwise it looks like this:
<img width="1666" alt="image" src="https://user-images.githubusercontent.com/2780116/181223590-79c6458a-6c0c-4e67-8f95-64ffd0bf0c52.png">